### PR TITLE
Revert deprecation of device flags

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @akihironitta @justusschock @kaushikb11 @awaelchli @borda @ananthsub @ninginthecloud @jjenniferdai

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -588,12 +588,6 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `Trainer.num_gpus` in favor of `Trainer.num_devices` when GPU is used ([#12384](https://github.com/PyTorchLightning/pytorch-lightning/pull/12384))
 
 
-- Deprecated `Trainer.ipus` in favor of `Trainer.num_devices` when IPU is used ([#12386](https://github.com/PyTorchLightning/pytorch-lightning/pull/12386))
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply CHANGELOG.md:591]: @rohitgr7 In our brief discussion earlier with Will and team we were just talking about the trainer arguments. The properties can be deprecated more freely as the impact and number of code changes on the user side is rare and small.

**Comment by awaelchli** [comment]
If we could just revert this addition [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/12156#discussion_r825228384), then that should be all I think. Couldn't find any others. We can simply resume efforts on #11040 after the release 😃

**Comment by DuYicong515** [comment]
> Users wouldn't understand the reason why we are deprecating `gpus`, `tpu_cores`, etc and moving towards `num_devices` unless they are enforced to use the `devices=x` logic.

The properties like `tpu_cores`, `num_processes`, `ipus` return the number of specific devices, and `gpus` return the user input of `gpus` or `devices` or what's decided by `auto`. I feel it's a bit confusing when the properties have the same name with the arguments, but return different things. 

Suggesting user to use device agnostic properties to access their device count or ids seems natural even we don't deprecate the arguments? The suggested properties are more general and have clear expectations on the return values from their names. In addition, if we decide they'll be removed eventually, I feel it's better to add deprecation message and prepare people early.

**Comment by daniellepintz** [comment]
Why not just do the deprecation of the constructor arguments in this release as well? https://github.com/PyTorchLightning/pytorch-lightning/pull/11040 is ready to go

**Comment by carmocca** [comment]
We are not comfortable doing this very impactful change for the community one day away from the release.

**Comment by awaelchli** [comment]
I'm closing this, as 1.6 has now been released. Happy deleting code everybody!

**Pull request closed without merging**